js/event.js

```diff
@@ -1,29 +1,119 @@
-"use strict"
+import { FormData } from "./database.js"
+import { MILISECOND_HOUR, getToday } from "./dateManipulation.js"
+import { selectedDate, storageState } from "./state.js"
+
+const TIMESLOT_LENGTH = 15
 
-const eventButtons_create = document.querySelectorAll('.create');
 const eventModal = document.querySelector("#eventModal")
+const eventForm = document.querySelector("#eventForm")
+const eventButtons_create = document.querySelectorAll('.create');
 const eventButton_cancel = document.querySelector("#event-cancel")
 const eventButton_save = document.querySelector("#event-save")
+const eventDate = document.querySelector('#event-date')
+const startTime = document.querySelector('#event-startTime')
+const endTime = document.querySelector('#event-endTime')
+
+//takes in string in form of 'HH:MM'
+const parseTimeString = (string) => {
+    const result = {}
+    result.hour = parseInt(string.slice(0, 2))
+    result.minutes = parseInt(string.slice(3, 5))
+    return result
+}
+
+const collectFormData = () => {
+
+    const formData = new FormData()
+
+    const inputData = {}
+    const inputs = document.querySelectorAll('[data-key]')
 
-const eventButtons = document.querySelectorAll('#eventModal button')
+    inputs.forEach(input => {
+        if(input.required && !input.value){
+            throw(`Required field (${input.dataset.key}) skipped`)
+        }
+        const key = input.dataset.key
+        const value = input.value
+        inputData[key] = value ? value : null
+    })
+
+    formData.title = inputData.title
+
+    const { startTime : startTimeString, endTime : endTimeString} = inputData
+
+    if(inputData.date && startTimeString && endTimeString){
+
+        const startTime = parseTimeString(startTimeString)
+        const startDate = new Date(inputData.date)
+        startDate.setHours(startTime.hour, startTime.minutes)
+        formData.startDate = Date.parse(startDate)
+    
+        const endTime = parseTimeString(endTimeString)
+        const endDate = new Date(inputData.date)
+        endDate.setHours(endTime.hour, endTime.minutes)
+        formData.endDate = Date.parse(endDate)
+
+        formData.duration = Math.ceil( (formData.endDate - formData.startDate) / (MILISECOND_HOUR/4) )
+        formData.timeSlot = Math.floor( startTime.minutes /  TIMESLOT_LENGTH)
+        formData.cellTimestamp = startDate.setMinutes(0).valueOf()
+    }
 
-eventButtons.forEach(btn => {
-    btn.addEventListener('click', (e) => e.preventDefault())
+    formData.setId()
+    formData.createdAt = getToday().valueOf()
+    formData.description = inputData.description
+
+    localStorage.setItem(formData.id, JSON.stringify(formData))
```

Comment:
You have a dedicated database.js source file for dealing with storing data. It should contain methods for storing and gathering data. It is not a very good idea to spread storing and gathering logic across the project. Imagine what would happen if you would need to change method of storage one day to something else. It will be much easier if you will find all functionality about it in database.js

Reply:
I would also suggest not to save data into index which is just number. It is better to give it some namespace because it is not clear what is stored there.

Possible solutions: you could have a key "forms" and store array of fields into it.

Also, you could have well-named setter and getter functions in database.js so that it would be clear what is being set (such as setFormData(formId, data))